### PR TITLE
Fixed DefinitionIsChildOfConstraintTest to use newer Symfony code

### PR DESCRIPTION
**Pull request by stloyd**

### CODE REVIEW

File: PhpUnit/DefinitionIsChildOfConstraint.php [hunk 1]
```diff
@@ -46,7 +47,13 @@ public function toString()
 
     private function evaluateDefinitionIsDecorator(Definition $definition, $returnResult)
     {
-        if (!($definition instanceof DefinitionDecorator)) {
+        if (!class_exists(ChildDefinition::class) && !$definition instanceof DefinitionDecorator) {
```
Comment (xabbuh) [inline PhpUnit/DefinitionIsChildOfConstraint.php]: could just be `if (!$definition instanceof ChildDefinition && !$definition instance of DefinitionDecorator)`

File: PhpUnit/DefinitionIsChildOfConstraint.php [hunk 2]
```diff
@@ -46,7 +47,13 @@ public function toString()
 
     private function evaluateDefinitionIsDecorator(Definition $definition, $returnResult)
     {
-        if (!($definition instanceof DefinitionDecorator)) {
+        if (!$definition instanceof ChildDefinition && !$definition instance of DefinitionDecorator) {
```
Comment (xabbuh) [inline PhpUnit/DefinitionIsChildOfConstraint.php]: `instanceof` instead of `instance of` :)

Comment (Nyholm) [reply PhpUnit/DefinitionIsChildOfConstraint.php]: Sorry. I need to use my glasses..

File: PhpUnit/DefinitionIsChildOfConstraint.php [hunk 3]
```diff
@@ -46,7 +47,13 @@ public function toString()
...
     private function evaluateDefinitionIsDecorator(Definition $definition, $returnResult)
     {
-        if (!($definition instanceof DefinitionDecorator)) {
+        if (!$definition instanceof ChildDefinition && !$definition instanceof DefinitionDecorator) {
+            if ($returnResult) {
+                return false;
+            }
+
+            $this->fail($definition, 'The definition has no parent service');
+        } elseif (!$definition instanceof ChildDefinition) {
```
Comment (xabbuh) [inline PhpUnit/DefinitionIsChildOfConstraint.php]: this `elseif` block can be removed (already covered above)

Comment (xabbuh) [reply PhpUnit/DefinitionIsChildOfConstraint.php]: and it makes tests with `DefinitionDecorator` instances fail

**Comment by Nyholm** [comment]
Thank you

**Comment by xabbuh** [comment]
looks like we also need to update the data provider to not include the data with the `DefinitionDecorator` class on Symfony 4

**Comment by Nyholm** [comment]
@stloyd I hope you do not mind that I made some changes to your branch

**Comment by stloyd** [comment]
@Nyholm Not a problem ;) I didn't had time to finish this fast, feel free to squash & merge ;)

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**